drivers/hmis/app/models/hmis/reminders/reminder_generator.rb

```diff
@@ -0,0 +1,206 @@
+        @mapped.fetch_values(*keys.map(&:to_sym))
+      end
+
+      def hoh_entry_date(enrollment)
+        @hoh_anniversary_by_household_id ||= enrollments
+          .group_by(&:household_id)
+          .map do |household_id, group|
+            hoh = group.detect { |e| e.RelationshipToHoH == 1 }
+            [household_id, hoh&.entry_date]
+          end
+          .to_h
+        @hoh_anniversary_by_household_id[enrollment.household_id]
+      end
+
+      def last_assessment_date(enrollment:, stages:, wip:)
+        # load all the assessments we might need
+        @all_assessments ||= Hmis::Hud::CustomAssessment
+          .where(enrollment_id: enrollments.map(&:enrollment_id), data_source_id: project.data_source_id)
+          .order(assessment_date: :desc)
+          .group_by(&:enrollment_id)
+
+        assessments = @all_assessments[enrollment.enrollment_id]
+        return unless assessments
+
+        stages = data_stages(stages)
+        found = assessments.detect do |r|
+          wip.include?(r.wip) && stages.include?(r.data_collection_stage)
+        end
+        found&.assessment_date
+      end
+
+      # Show reminder if ANY client in the household is missing an Annual Assessment within the
+      # range when the annual is due, and today is on or after the start of that range.
+      def annual_assessment_reminder(enrollment)
+        hoh_entered_on = hoh_entry_date(enrollment)
+        return unless hoh_entered_on
+
+        hoh_entered_on = normalize_yoy_date(hoh_entered_on)
+        window = 30.days
+        # not due for an assessment yet
+        return if today < (hoh_entered_on + (1.year - window))
+
+        hoh_anniversary = hoh_entered_on.change(year: today.year)
+        start_date = hoh_anniversary - window
+        due_date = hoh_anniversary + window
+
+        # a relevant assessment ocurred.
+        # FIXME: maybe we don't include assessments that occur after end_date? This might
+        # encourage people to back-date assessments.
+        last_assessed_on = last_assessment_date(enrollment: enrollment, stages: [:annual_assessment], wip: [false])
+        return if last_assessed_on && last_assessed_on >= start_date
+
+        new_reminder(
+          topic: ANNUAL_ASSESSMENT_TOPIC,
+          due_date: due_date,
+          enrollment: enrollment,
+        )
+      end
+
+      # Show reminder if a client in the household has turned 18 since the entry date, and they
+      # have not had either an update or an annual since turning 18.
+      def aged_into_adulthood_reminder(enrollment)
+        age_of_majority = 18
+        client = enrollment.client
+        return unless client.DOB
+
+        entry_date = enrollment.entry_date
+        adulthood_birthday = normalize_yoy_date(client.DOB) + age_of_majority.years
+        # the client was already an adult on entry date
+        return if adulthood_birthday <= entry_date
```

Comment:
The reminder shouldn't be shown unless the client is 18+. No point in showing it earlier because it shouldn't be completed until the client has aged into adulthood

```suggestion
        # client is still <18
        return if adulthood_birthday > Date.current
        # the client was already an adult on entry date
        return if adulthood_birthday <= entry_date
```

Reply:
Ah oops, thanks for the catch! I also fixed the tests that were checking the wrong reminder topic 🤦